pkg/controller/service/service.go

```diff
@@ -13,25 +16,48 @@ func RenderServices(req router.Request, resp router.Response) error {
 	// reset, should get repopulated
 	svcInstance.Status.Endpoints = nil
 
-	objs, _, err := services.ToK8sService(req, svcInstance)
-	if err != nil {
-		return err
+	var svcList []*v1.ServiceInstance
+	http2Ports := ports.ByProtocol(svcInstance.Spec.Ports, true, v1.ProtocolHTTP2)
+	otherPorts := ports.ByProtocol(svcInstance.Spec.Ports, false, v1.ProtocolHTTP2)
+	if len(http2Ports) > 0 {
+		svcCopy := svcInstance.DeepCopy()
+		svcCopy.Spec.Ports = http2Ports
+		svcCopy.Name = fmt.Sprintf("%s-%s", svcInstance.Name, v1.ProtocolHTTP2)
```

Comment:
I am agreeing with @StrongMonkey here. Creating an additional `ServiceInstance` with a different name may cause issues with other parts of the system. For example, if something depends on the service, but the service has a different name. Or, the user may create a service that results in a `ServiceInstance` being created with the same name you are creating here.

Would it be possible to render the `corev1.Service` with the special annotations instead of creating an additional `ServiceInstance`?

Reply:
The `ServiceInstance` is only created as a temporary copy for our purposes so that we can pass the same value into the function logic. We don't actually return the temporarily created `ServiceInstance` object, we just use it to create a new `corev1.Service`.